microprofile/WFLY-10711_health_check_extension.adoc

```diff
@@ -0,0 +1,82 @@
+= [WFLY-10711] Support Eclipse MicroProfile Health
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:keywords:          observability,microprofile,health,openshift
+
+== Overview
+
+In order to improve Observability of WildFly on OpenShift, WildFly needs to expose health check probes
+so that the container can query whether WildFly and its deployed applications are healthy or not.
+
+This will be implemented by supporting Eclipse MicroProfile Health 1.0 for applications deployed in WildFly.
+The integration uses the smallrye-health component to provide the MicroProfile Health implementation.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-10711[WFLY-10711]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1029[EAP7-1029]
+* https://issues.jboss.org/browse/CLOUD-2730[CLOUD-2730] - Implement EAP Health Check so that it can be consumed by OpenShift Console
+* https://issues.jboss.org/browse/WFLY-10522[WFLY-10522] - MicroProfile Config extension for WildFly.
+** There is no direct relation between MicroProfile Health and MicroProfile Config but there should be consistency in their extensions (subsystem names, etc.)
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:tremes@redhat.com[Tomas Remes]
+
+=== Affected Projects or Components
+
+The implementation of Eclipse MicroProfile Health is provided by https://github.com/smallrye/smallrye-health[smallrye-health].
+
+=== Other Interested Projects
+
+* A https://github.com/jmesnil/wildfly-microprofile-health[prototype of a WildFly extension] for MicroProfile Health was developed during the development of MicroProfile Health 1.0.
+
+== Requirements
+
+* Support Eclipse MicroProfile Health 1.0 for applications deployed in WildFly.
+* Provide a HTTP endpoint for the https://github.com/eclipse/microprofile-health/blob/master/spec/src/main/asciidoc/protocol-wireformat.adoc[health wireformat]
```

Comment:
What are the requirements re: the configurability of that endpoint?

We've discussed exposing this over the management HTTP interface, so if that's all we are doing then there's probably no configurability at all, since it can just be integrated via a requirement for the static capability exposed by that resource.  But the proposal be clear that the exposure would only be over HTTP management and that having that interface in the config is required.

Reply:
@bstansberry I've clarified in the proposal.

Note that I've changed my mind:
* the subsystem will expose a `check` operation to query the healthiness of the application server (this way, we can locally query the healthiness without exposing anything more on HTTP(S)).
* HTTP endpoint is optional. If it is enabled, it requires the capability for `org.wildfly.management.http-interface`